llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -8904,6 +8897,12 @@ VPRecipeBuilder::tryToCreatePartialReduction(Instruction *Reduction,
       isa<VPPartialReductionRecipe>(BinOpRecipe))
     std::swap(BinOp, Accumulator);
 
+  VPValue *Mask = getBlockInMask(Reduction->getParent());
+  if (Mask) {
+    VPValue *Zero =
+        Plan.getOrAddLiveIn(ConstantInt::get(Reduction->getType(), 0));
+    BinOp = Builder.createSelect(Mask, BinOp, Zero, Reduction->getDebugLoc());
+  }
```

Comment:
Not sure, but should this be (or similar):
```suggestion
  if (CM.isPredicatedInst(Reduction)) {
    VPValue *Mask = getBlockInMask(Reduction->getParent());
    VPValue *Zero =
        Plan.getOrAddLiveIn(ConstantInt::get(Reduction->getType(), 0));
    BinOp = Builder.createSelect(Mask, BinOp, Zero, Reduction->getDebugLoc());
  }
```
Just because I see a `Trying to access mask for block without one.` assert in `getBlockInMask()`, and I wonder if it's possible to hit that somehow.

Reply:
Done.
The assert wasn't getting hit for the non-predicated tests, as they would return a mask of 0 rather than no mask at all. But I've added the extra condition for more safety. It needed to be a slightly different function inside the condition as this one was not working as required.